src/cats/SlideAnimation.tsx

```diff
@@ -0,0 +1,59 @@
+import { useEffect, useState } from "react";
+import clsx from "clsx";
+import styles from "./SlideAnimation.module.css";
+
+enum Direction {
+  Previous,
+  Next,
+}
+
+type SlideAnimationPropsType = {
+  child: JSX.Element;
+  direction: Direction;
+};
+
+function SlideAnimation({ child, direction }: SlideAnimationPropsType) {
+  const [slideChildren, setSlideChildren] = useState<Array<JSX.Element>>([]);
+
+  useEffect(() => {
+    setSlideChildren((prevChildren: Array<JSX.Element>) => {
+      const prevChild = prevChildren.slice(
+        ...(direction === Direction.Next ? [-1] : [0, 1])
+      );
+
+      return prevChild[0]?.key === child.key
+        ? prevChildren
+        : direction === Direction.Next
+        ? prevChild.concat([child])
+        : [child].concat(prevChild);
+    });
+  }, [child, direction]);
+
+  const onAnimationEnd = () => {
+    setSlideChildren([child]);
+  };
+
+  return (
+    <div className={styles.slideAnimation}>
+      {slideChildren.length === 2
+        ? [
+            slideChildren[0],
```

Comment:
You might want to make sure this doesn't re-mount the child that is coming in when the animation completes. It's probably not worth the cost just to remove the wrapper.